File: cglib/src/main/java/com/chogoon/cglib/OnListChangedCallback.java
package com.chogoon.cglib;

import androidx.databinding.ObservableList;

public abstract class OnListChangedCallback extends ObservableList.OnListChangedCallback {
    public OnListChangedCallback() {
        super();
    }

    @Override
    public void onChanged(ObservableList sender) {

    }

    @Override
    public void onItemRangeChanged(ObservableList sender, int positionStart, int itemCount) {

    }

    @Override
    public void onItemRangeInserted(ObservableList sender, int positionStart, int itemCount) {

    }

    @Override
    public void onItemRangeMoved(ObservableList sender, int fromPosition, int toPosition, int itemCount) {

    }

    @Override
    public void onItemRangeRemoved(ObservableList sender, int positionStart, int itemCount) {

    }
}

File: cglib/src/main/java/com/chogoon/cglib/OnListChangedCallBack.kt
package com.chogoon.cglib

import androidx.databinding.ObservableList
import androidx.databinding.ObservableList.OnListChangedCallback

abstract class OnListChangedCallBack<T : ObservableList<*>> : OnListChangedCallback<T>() {

    override fun onChanged(sender: T) {

    }

    override fun onItemRangeRemoved(sender: T, positionStart: Int, itemCount: Int) {

    }

    override fun onItemRangeMoved(sender: T, fromPosition: Int, toPosition: Int, itemCount: Int) {

    }

    override fun onItemRangeInserted(sender: T, positionStart: Int, itemCount: Int) {

    }

    override fun onItemRangeChanged(sender: T, positionStart: Int, itemCount: Int) {}
}